modules/acm_sku/src/Entity/SKU.php

```diff
@@ -444,76 +443,12 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
     // Check if we have additional fields to be added as base fields.
     if (!empty($additionalFields) && is_array($additionalFields)) {
       foreach ($additionalFields as $machine_name => $field_info) {
-        // Initialise the field variable.
-        $field = NULL;
 
         // Showing the fields at the bottom.
         $weight = $defaultWeightIncrement + count($fields);
 
-        switch ($field_info['type']) {
-          case 'attribute':
-          case 'string':
-            $field = BaseFieldDefinition::create('string');
-
-            if ($field_info['visible_view']) {
-              $field->setDisplayOptions('view', [
-                'label' => 'above',
-                'type' => 'string',
-                'weight' => $weight,
-              ]);
-            }
-
-            if ($field_info['visible_form']) {
-              $field->setDisplayOptions('form', [
-                'type' => 'string_textfield',
-                'weight' => $weight,
-              ]);
-            }
-            break;
-
-          case 'text_long':
-            $field = BaseFieldDefinition::create('text_long');
-
-            if ($field_info['visible_view']) {
-              $field->setDisplayOptions('view', [
-                'label' => 'hidden',
-                'type' => 'text_default',
-                'weight' => $weight,
-              ]);
-            }
-
-            if ($field_info['visible_form']) {
-              $field->setDisplayOptions('form', [
-                'type' => 'text_textfield',
-                'weight' => $weight,
-              ]);
-            }
-            break;
-        }
-
-        // Check if we don't have the field type defined yet.
-        if (empty($field)) {
-          throw new \RuntimeException('Field type not defined yet, please contact TA.');
-        }
-
-        // We want to allow field labels to be translatable.
-        // Since we try to do this dynamically, we need to use t() with
```

Comment:
@gupta-ashutosh5 please copy this comment to new place

Reply:
done